### PR TITLE
Rich message text and image buttons

### LINKED ISSUE #6786: Rich Messages and Payloads for Buttons and Cards
It would be great (especially within chatbot workflows) to support interactive message buttons (like e.g. in slack https://rocket.chat/docs/developer-guides/rest-api/chat/postmessage#attachments-detail). As the slack syntax is quite straight forward, it can look analogously.
```
{
  // some properties
    "attachments": [
      {
       // some properties
       "callback_id": "question_abc_123",
       "actions": [
                {
                    "name": "question_abc",
                    "text": "Yes",
                    "type": "button",
                    "value": "yes"
                },
                 {
                    "name": "question_abc",
                    "text": "No",
                    "type": "button",
                    "value": "no"
                }
            ]
        }]
}
```
The answer could look like:
```
{
  // some properties
  "message": {
    //some properties
    "actions": [
      {
         "name": "question_abc",
         "text": "Yes",
         "type": "button",
         "value": "yes"
       }
    ],
    "callback_id": "question_abc_123",
  }
}
```

Rocket.Chat Version: 0.55.1
Running Instances: 
DB Replicaset OpLog:
Node Version:

### PR DESCRIPTION
**Pull request by ubarsaiyan**
Demo: https://drive.google.com/file/d/1kbNLqs2g_v5PLVJWro0Pz-d0e6Wjbp1f/view?usp=sharing

Closes #6786

This PR aims to pass all the test specified in https://github.com/WideChat/Rocket.Chat.Android/wiki/Phase-1--Rich-Messaging-Acceptance-Test

Currently, all tests should pass except 7 and 8. :)

### CODE REVIEW

File: packages/rocketchat-message-action/client/stylesheets/messageAction.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.attachment {
+
+	& .action {
+		margin-top: 2px;
+	}
+
+	& .text-button {
+		color: #2c2d30;
+		min-width: 0;
+    	max-width: 220px;
```
Comment (rodrigok) [inline packages/rocketchat-message-action/client/stylesheets/messageAction.css]: Can you fix the indentation here?

**Comment by ubarsaiyan** [comment]
@rodrigok Fixed indentation and rebased to latest develop

**Comment by rodrigok** [comment]
@ubarsaiyan can you add the new properties to the schema check here https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-lib/server/functions/sendMessage.js#L28 ?

**Comment by ubarsaiyan** [comment]
@rodrigok I have added the new properties, pls check :)

**Comment by ivanmpk** [comment]
Hi, I use Rocket 0.70.4. How to add this to my server? If necessary reinstall? or copy files into my path instalation? 

Thanks!

**Comment by ubarsaiyan** [comment]
Hi @ivanmpk 
Look at this wiki to send the correct JSON for the rich messages. https://github.com/WideChat/Rocket.Chat.Android/wiki/Phase-1--Rich-Messaging-Acceptance-Test

**Comment by ivanmpk** [comment]
Hi @ubarsaiyan thanks for your quick response.
Ok on wiki I see how to send JSON but how to load the files .js and .html on my server. Only copy files? what files?

**Comment by ubarsaiyan** [comment]
@ivanmpk Just normally install the server for the richmessage functionality. This would only enable richmessage messages to be parsed on the client. To send a richmessage you would need to connect a suitable bot.

**Comment by ivanmpk** [comment]
@ubarsaiyan thanks again. How to install rocket with richmessage funciontality?

**Comment by ubarsaiyan** [comment]
@ivanmpk Just normal installation as mentioned in the README. The PR was merged in the version 0.69.0

**Comment by ivanmpk** [comment]
Excellent! Thank you!

**Comment by ivanmpk** [comment]
Hi, the button_alignment its not working for me. Always the buttons are vertical. 
Do you have estimated when test 7 and 8 will be success? I need "msg_in_chat_window: false"

**Comment by ubarsaiyan** [comment]
@ivanmpk Are you sending ` "button_alignment": "horizontal"` outside the `actions` and within the `attachments` as mentioned in the wiki? If yes, then please open an issue describing it in more detail. There is no progress that I know of regarding the tests 7 and 8.

**Comment by ubarsaiyan** [comment]
@ivanmpk I don't know. Maybe ask in #dev channel @ open.rocket.chat?

**Comment by buerlyu** [comment]
@ubarsaiyan Hi, I'm currently working on a contact center demo by using dialogflow based chatbot and the livechat in rocketchat. I try to create a button but can only get a blank message in the livechat.
```
            let msgData = {
              channel: roomId,
              //rid: roomId,
              text: "text button with msg in chat window",
              attachments: [
                {
                  actions: [
                    {
                      type: "button",
                      text: "Say hello in chat window?",
                      msg: "hello in chat window",
                      msg_in_chat_window: true
                    }
                  ]
                }
              ]
            };
            robot.adapter.customMessage(msgData);
```
image

**Comment by ubarsaiyan** [comment]
@bizzbyster perhaps you can shed some light to @buerlyu on the current state of the rich messages.

**Comment by matikucharski** [comment]
Hi. Where are rich messages documented? I cannot find a word on it in RC documentation on https://developer.rocket.chat/

**Comment by cicada-chen** [comment]
where is the doc describe this feature?

**Pull request merged**